hugo/content/guides/formatting.md

```diff
@@ -0,0 +1,103 @@
+    lsp: {
+        Formatter: () => new CustomModelFormatter()
+    }
+};
+```
+
+The entry point for the formatter is the abstract `format(AstNode)` method. The `AbstractFormatter` calls this method for every node of our model.
+To perform custom formatting for every type of node, we will use pattern matching.
+In the following example, we will take a closer look at a formatter for the [domain-model](https://github.com/langium/langium/tree/main/examples/domainmodel) language.
+In particular, we will see how we can format the root of our model (`DomainModel`) and each nested element (`Entity` and `PackageDeclaration`).
+
+To format each node, we use the `getNodeFormatter` method of the `AbstractFormatter`. The resulting generic `NodeFormatter<T extends AstNode>` provides us with methods to select specific parts of a parsed `AstNode` such as properties or keywords.
+
+Once we have selected the nodes of our document that we are interested in formatting, we can start applying a specific formatting.
+Each formatting option allows to prepend/append whitespace to each note.
+The `Formatting` namespace provides a few predefined formatting options which we can use for this.
+ 
+We first start off by formatting the `Domainmodel` element of our DSL.
+It is the root node of every document and just contains a list of other elements.
+These elements need to be realigned to the root of the document in case they are indented.
+We will use the `Formatting.noIndent` options for that:
+
+```ts
+if (ast.isDomainmodel(node)) {
+    // Create a new node formatter
+    const formatter = this.getNodeFormatter(node);
+    // Select a formatting region which contains all children
+    const nodes = formatter.nodes(...node.elements);
+    // Prepend all these nodes with no indent
+    nodes.prepend(Formatting.noIndent());
+}
+```
+
+Our other elements, namely `Entity` and `PackageDeclaration`, can be arbitrarily deeply nested, so using `noIndent` is out of the question for them.
+Instead we will use `indent` on everything between the `{` and `}` tokens. The formatter internally keeps track of the current indentation level:
+
+```ts
+if (ast.isEntity(node) || isPackageDeclaration(node)) {
+    const formatter = this.getNodeFormatter(node);
+    const bracesOpen = formatter.keyword('{');
+    const bracesClose = formatter.keyword('}');
+    // Add a level of indentation to each element
+    // between the opening and closing braces.
+    // This even includes comment nodes
+    formatter.interior(bracesOpen, bracesClose).prepend  (Formatting.indent());
+    // Also move the newline to a closing brace
+    bracesClose.prepend(Formatting.newLine());
+    // Surround the name property of the element
+    // With one space to each side
+    formatter.property("name").surround(Formatting.oneSpace());
+}
+```
+
+Note that most predefined `Formatting` methods accept additional arguments which make the resulting formatting more lenient.
```

Comment:
Do you mind to tell more about these methods? Yes, it's a guide and not a documentation, but maybe it's a nice to see a couple more examples of API usage.